README.md

```diff
@@ -1,12 +1,52 @@
 # Self-healing Apache Zookeeper ECS cluster
 
 ## General information
-This repo contains terraform code and docker file which will allow you to create self-healing Apache Zookeeper cluster into AWS ECS
+The repository is a fork from https://github.com/dethmix/self-healing-zookeeper. self-healing-zookeeper deploys each instance of the zookeeper cluster as a separate service. It then uses ECS cloud map to discover its peers in the cluster.
 
+With this repo we aimed to provide this with lesser number of moving parts. Instead of keeping the cluster nodes as part of separate services, we combined them into 1 single service. We used the underlying behavior of how nodes are scheduled in a service when there are multiple subnets. If the service has multiple subnets, each replica that comes up for the service will attach to the subnet that has no instances in that subnet.
 ## Usage
 Terraform code can be found in `terraform` directory, just update variable.tf file with values related to your AWS VPC and execute `terraform plan && terraform apply`
- 
-Dockerfile for building custom Apache Zookeeper image can be found in `docker` directory. It contains only few custom changes in comparing to the original Apache Zookeeper dockerfile. The main one is generation of `myid` from ECS service name.
 
-## Blog post
-More details about this setup can be found at https://tech.smartling.com/self-healing-apache-zookeeper-cluster-470b248ccb12
+
+## Infrastructure
+
+The infrastructure required to implement this solution are the following:
+
+Number of subnets equal to the number of nodes desired to be in the cluster
+Note: Each subnet houses a single instance of a cluster.
+
+When the container comes up in the respective subnets, there are 2 requirements for that container to be part of the zookeeper cluster.
+- The container needs to update its IP with correct domain in route53. We have provided that intelligence to the container where they register the domain appropriately with route53.
+- The container needs to assign the correct zookeeper id to itself to be part of the zookeeper cluster
+
+We automated the Zookeeper node deployment by adding DNS names in the configs instead of IP addresses while defining the members of the zookeeper cluster. This ensures that we don't have to change the IP address again when we want to deploy zookeeper.
+
+e.g
+```
+ZOO_HOSTED_ZONE=T24WYUCZ179
+```
+
+The domain name convention should be configured in a way that it coincides with whatever value the subnet (third octet of the subnet cidr) has. 
```

Comment:
Every zookeeper instance that comes up as a cluster needs to have a predetermined ID. We use the third octet of the subnet CIDR to assign the ID value for the zookeeper instance.